darts/models/tcn_model.py

```diff
@@ -84,7 +84,7 @@ def __init__(self,
         if weight_norm:
             self.conv1, self.conv2 = nn.utils.weight_norm(self.conv1), nn.utils.weight_norm(self.conv2)
 
-        if nr_blocks_below == 0 or nr_blocks_below == num_layers - 1:
+        if nr_blocks_below in {0, num_layers - 1} and input_dim != output_dim:
```

Comment:
Actually do you think we could simplify this to
```suggestion
        if  input_dim != output_dim:
```

Reply:
Agree! Looks super simple and clear